lib/chef/provisioning/docker_driver/driver.rb

```diff
@@ -103,8 +112,8 @@ def build_container(machine_spec, machine_options)
         "#{source_repository}:#{source_tag}"
       else
         target_repository = 'chef'
-        target_tag = machine_spec.name
-
+        target_tag = machine_spec.location['container_name'] || machine_spec.name
```

Comment:
Attribute `location` seems obsolete. I would suggest to use `reference` instead.